parser/parser.coffee

```diff
@@ -0,0 +1,118 @@
+scanner = require '../scanner/scanner'
+CustomError = require '../error/custom_error'
+
+Program = require '../entities/01_program'
+Block = require '../entities/02_block'
+ForStatement = require '../entities/03_for_statement'
+WhileStatement = require '../entities/03_while_statement'
+
+StartTokens = require './start_tokens'
+
+tokens = []
```

Comment:
The correct terminology is that top-level variables in a file are "**local** to a module." Don't say "global here." JavaScript only has global variables and local variables. Local variables are local to a function. The way Node works is that for each file you require (or load, if it is a script), it wraps the content of the file in a function. So everything at the top-level of a file actually becomes a local variable to the module's wrapped function. The only ways to make a truly global variable in Node is (1) In JavaScript, forget the var, or (2) in JS or CS, assign to a property of `global`.

Anyway, as to your question of whether you want classes or not, it depends on how much you like OOP. Do you want to see your parser as a living, breathing thing? Then make a class and instantiate it. Don't make a singleton class or a utility class; singletons are considered anti-patterns these days. You can just make a module, though. We'll talk about this in class if you like.